src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx

```diff
@@ -202,23 +207,27 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
                     variant="outlined"
                     error={!!errors.unitName}
                     helperText={errors.unitName?.message}
+                    key={field.value}
+                    onChange={(event) => {
+                      const selectedValue = event.target.value;
+                      if (selectedValue === 'createUnit') {
+                        const unitName = prompt('Enter Unit Name');
+                        if (unitName) {
+                          createUnit(unitName);
+                        } else {
+                          event.target.value = field.value === undefined ? '' : field.value;
```

Comment:
Unit doesn't seem to be working for me. It still sets it to + Create Unit when I hit cancel